packages/taco/src/conditions/context/context.ts

```diff
@@ -222,16 +219,19 @@ export class ConditionContext {
         ERR_AUTH_PROVIDER_NOT_NEEDED_FOR_CONTEXT_PARAM(contextParam),
       );
     }
+    const expectedType = EXPECTED_AUTH_PROVIDER_TYPES[contextParam];
+    const isValid = Array.isArray(expectedType)
+      ? expectedType.some(type => authProvider instanceof type)
+      : authProvider instanceof expectedType;
 
-    if (!(authProvider instanceof EXPECTED_AUTH_PROVIDER_TYPES[contextParam])) {
+    if (!isValid) {
```

Comment:
Given the above comment about the types, then we can do the following:
```typescript
  public addAuthProvider(contextParam: string, authProvider: AuthProvider) {
    if (!(contextParam in EXPECTED_AUTH_PROVIDER_TYPES)) {
      throw new Error(
        ERR_AUTH_PROVIDER_NOT_NEEDED_FOR_CONTEXT_PARAM(contextParam),
      );
    }
    const expectedTypes = EXPECTED_AUTH_PROVIDER_TYPES[contextParam];
    if (!expectedTypes.some((type) => authProvider instanceof type)) {
      throw new Error(
        ERR_INVALID_AUTH_PROVIDER_TYPE(contextParam, typeof authProvider),
      );
    }

    this.authProviders[contextParam] = authProvider;
  }
```

Reply:
yeah it definitely simplifies things if we just always make things arrays, even if they only contain one item